### PR TITLE
Support moreh logsoftmax

### PR DESCRIPTION
**Pull request by hschoi4448**
- refactoring moreh softmax and softmin to receive an output tensor as function input.
- implement moreh logsoftmax
- change comp_pcc to comp_allclose_and_pcc 

logsoftmax: https://pytorch.org/docs/1.7.1/nn.functional.html#torch.nn.functional.log_softmax

### CODE REVIEW

**Review by nadongguri — APPROVED** [review]
LGTM.

**Review by arakhmati — APPROVED** [review]
LGTM.

And you don't need to do it now but you can start using optional input tensors that we created for your ops: https://tenstorrent-metal.github.io/tt-metal/latest/ttnn/dependencies/tt_lib.html#new-device-operation-with-optional-output-tensors
